bigquery-storage/api/BigQueryStorage.Samples/AppendRowsPending.cs

```diff
@@ -0,0 +1,141 @@
+{
+    /// <summary>
+    /// This code sample demonstrates how to write records in pending mode
+    /// Create a write stream, write some sample data, and commit the stream to append the rows.
+    /// The CustomerRecord proto used in the sample can be seen in Resources folder and generated C# is placed in Data folder in the Samples directory
+    /// </summary>
+    /// <param name="projectId">The Google cloud project Identifier</param>
+    /// <param name="datasetName">The BigQuery Dataset name</param>
+    /// <param name="tableName">The BigQuery Table name</param>
+    /// <returns>The task</returns>
+    public async Task AppendRowsPendingAsync(string projectId, string datasetName, string tableName)
+    {
+        BigQueryWriteClient bigQueryWriteClient = await BigQueryWriteClient.CreateAsync();
+        // Initialize a write stream for the specified table.
+        // When creating the stream, choose the type. Use the PENDING type to wait
+        //until the stream is committed before it is visible. See:
+        // https://cloud.google.com/bigquery/docs/reference/storage/rpc/google.cloud.bigquery.storage.v1#google.cloud.bigquery.storage.v1.WriteStream.Type
+        WriteStream stream = new WriteStream { Type = WriteStream.Types.Type.Pending };
+        TableName table = TableName.FromProjectDatasetTable(projectId, datasetName, tableName);
+        
+        stream = await bigQueryWriteClient.CreateWriteStreamAsync(table, stream);
+
+        // Initialize streaming call, retrieving the stream object
+        BigQueryWriteClient.AppendRowsStream rowAppender = bigQueryWriteClient.AppendRows();
+
+        // Sending requests and retrieving responses can be arbitrarily interleaved
+        // Exact sequence will depend on client/server behavior
+        // Create task to do something with responses from server
+        Task appendResultsHandlerTask = Task.Run(async () =>
+        {
+            AsyncResponseStream<AppendRowsResponse> appendRowResults = rowAppender.GetResponseStream();
+            while (await appendRowResults.MoveNextAsync())
+            {
+                AppendRowsResponse responseItem = appendRowResults.Current;
+                // Do something with responses
+                Console.WriteLine($"Response Append Result is {responseItem.AppendResult}");
+            }
+            // The response stream has completed
+        });
+
+        // Populate the data
+        // Create a batch of row data by appending proto2 serialized bytes to the
+        // SerializedRows repeated field.
+        ProtoData protoData = new ProtoData
+        {
+            // This is a TEMPORARY WORKAROUND via extension method using reflection till the feature request
+            // and PR for "ToProto()" in Protobuf library gets released.
+            WriterSchema = new ProtoSchema { ProtoDescriptor = CustomerRecord.Descriptor.ToProto() },
+            Rows = new ProtoRows
+            {
+                SerializedRows =
```

Comment:
I would be tempted to separate "initializing the data" from "initializing the request". That's likely to mirror actual customer code more (where the data would probably be passed into the method). Something like this (but with appropriate comments):

```csharp
List<CustomerRecord> records = new List<CustomerRecord>
{
    new CustomerRecord { CustomerNumber = 1, CustomerName = "Alice"},
    new CustomerRecord { CustomerNumber = 2, CustomerName = "Bob"}
};

ProtoData protoData = new ProtoData
{
    WriterSchema = new ProtoSchema { ProtoDescriptor = CustomerRecord.Descriptor.ToProto() },
    Rows = new ProtoRows { SerializedRows = { records.Select(r => r.ToByteString()) } }
};
```

Reply:
Done.